File: Neetcode/LongestConSeq.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        hashset = set(nums)
        max_seq = 0
        for n in hashset:
            if n-1 not in hashset:
                curr_seq = 0
                while n+curr_seq in hashset:
                    curr_seq += 1
                max_seq = max(max_seq, curr_seq)
        return max_seq

File: Neetcode/LongestConSeq.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> numset;
        int count = 0, ans=0;
        for (int n:nums){
            numset.insert(n);
        }
        for(int n:nums){
            count += 1;
            if (numset.count(n-1) == 0){
                while (numset.count(n+1) != 0){
                    count += 1;
                    n += 1;
                }
            }
            ans = max(count, ans);
            count = 0;
        }
        return ans;
    }
};
